tests/CMakeLists.txt

```diff
@@ -0,0 +1,7 @@
+enable_testing()
+find_package(GTest REQUIRED)
+
+add_executable(cps-tests loader.cpp utils.cpp version.cpp)
+target_link_libraries(cps-tests PRIVATE libcps)
+target_link_libraries(cps-tests PRIVATE GTest::gtest GTest::gtest_main)
+add_test(CpsTests cps-tests)
```

Comment:
I think it's at least noting with `TODO` style comments anything that the meson build files are able to do that the CMake isn't (like running the python based tests).

Reply:
This is an unintentional omission. I'll add it

Reply:
Help wanted: how do I make cmake consume TAP output? Current attempt that does not properly consume TAP output:
```cmake
# Integration tests
find_package(
  Python 3.11
  COMPONENTS Interpreter
  REQUIRED)
add_test(
  NAME pkg-config-compatibility-tests
  COMMAND
    ${CMAKE_COMMAND} -E env CPS_PATH="${CMAKE_CURRENT_SOURCE_DIR}/cases"
    ${PYTHON_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/runner.py
    $<TARGET_FILE:cps-config> ${CMAKE_CURRENT_SOURCE_DIR}/cases.toml)

```

Reply:
@dcbaker Google seems to indicate that CMake does not have a consumer for TAP tests. I'm reading from the TAP spec that it explicitly disregard exit code? Would it break anything if I make `runner.py` return non-zero on failure?

Reply:
I implemented this change. Let me know if this breaks anything. If it does I'll look for another solution.

Reply:
It should be fine as long as it returns 0 for success and 1 for failure. TAP does have some return code stuff, but it's not documented outside of perl code that I could find. I'm also open to switching away from TAP, it was just pretty easy to implement and Meson could consume it easily

Reply:
I'm honestly not sure what protocol CMake supports best. It seems to me that CMake expects a different command being run for each test case. This would require either hard-coding a list of testcases in CMakeLists or have first party support for a testing framework through something like `gtest_discover_tests`. Or of course treat the entire suite as one test case. None of the three sounds optimal to me.

I think having it return 1 on failure is the route that makes most sense currently.